pontoon/base/views.py

```diff
@@ -482,6 +482,41 @@ def unreject_translation(request):
     })
 
 
+@utils.require_AJAX
+@login_required(redirect_field_name='', login_url='/403')
+@transaction.atomic
+def delete_translation(request):
+    """Delete given translation."""
+    try:
+        t = request.POST['translation']
+    except MultiValueDictKeyError as e:
+        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+
+    translation = get_object_or_404(Translation, pk=t)
+    project = translation.entity.resource.project
+    locale = translation.locale
+
+    # Read-only translations cannot be deleted
+    if utils.readonly_exists(project, locale):
+        return HttpResponseForbidden(
+            "Forbidden: This string is in read-only mode"
+        )
+
+    # Only privileged users or authors can delete translations
+    if not translation.rejected or not (
```

Comment:
[nit] `get_object_or_404` can take a QuerySet as the first parameter. That's totally subjective but it could be more easier to read/refactor it later.
e.g.
```
translation = get_object_or_404(Translation.objects.rejected, pk=4)
if not(request.user.can_translate(locale,project) or request.user == translation.user):
    ...
```

Reply:
We'd need a manager for `Translation.objects.rejected` to work, though - otherwise this is a an `AttributeError`.